### PR TITLE
chore: move to github actions

### PR DESCRIPTION
**Pull request by okwasniewski**
This PR adds new architecture related CI jobs and moves CI to github actions.

### TODO
- [x] remove CircleCI in GH settings 
- [x] cancel previous job if it is being processing 

https://github.com/callstack/react-native-pager-view/actions/runs/3410171591

Screenshot 2022-11-07 at 13 11 06

- [x] https://reactnative.dev/docs/build-speed

### CODE REVIEW

File: .github/workflows/PagerCI.yml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+jobs:
+  install-dependencies:
+    name: Install dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: yarn
```
Comment (krozniata) [inline .github/workflows/PagerCI.yml:14]:
```suggestion
        run: yarn install
```
to keep consistency with other steps

File: .github/workflows/PagerCI.yml [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      - name: Cache example node modules
+        id: cache-example-npm
+        uses: actions/cache@v3
+        env:
+          cache-name: cached-example-npm-deps
+        with:
+          path: example/node_modules
+          key: ${{ hashFiles('./example/yarn.lock') }}-${{ hashFiles('./yarn.lock') }}
+
+      - name: Install required dependencies on cache miss (npm)
```
Comment (krozniata) [inline .github/workflows/PagerCI.yml:31]:
```suggestion
      - name: Install required example dependencies on yarn.lock change
```

File: .github/workflows/PagerCI.yml [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      - name: Cache fabric example node modules
+        id: cache-fabricexample-npm
+        uses: actions/cache@v3
+        env:
+          cache-name: cached-fabric-npm-deps
+        with:
+          path: fabricexample/node_modules
+          key: ${{ hashFiles('./fabricexample/yarn.lock') }}-${{ hashFiles('./yarn.lock') }}
+
+      - name: Install required dependencies on cache miss (npm)
```
Comment (krozniata) [inline .github/workflows/PagerCI.yml:45]:
```suggestion
      - name: Install required fabric example dependencies on yarn.lock change
```

File: .github/workflows/ios.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        run: pod install
+        working-directory: fabricexample/ios
+
+      - name: Build iOS (Fabric)
+        run: |
+          device_name='iPhone 13'
+          device=$(xcrun simctl list devices "${device_name}" available | grep "${device_name} (")
+          re='\(([-0-9A-Fa-f]+)\)'
+          [[ $device =~ $re ]] || exit 1
+          xcodebuild -workspace FabricExample.xcworkspace -scheme FabricExample -destination "platform=iOS Simulator,id=${BASH_REMATCH[1]}" CODE_SIGNING_ALLOWED=NO COMPILER_INDEX_STORE_ENABLE=NO build
```
Comment (krozniata) [inline .github/workflows/ios.yml:112]:
```suggestion
        run: xcodebuild -scheme FabricExample -workspace FabricExample.xcworkspace -sdk iphonesimulator -configuration Debug
```

File: .github/workflows/ios.yml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        run: pod install
+        working-directory: example/ios
+
+      - name: Build iOS
+        run: |
+          device_name='iPhone 13'
+          device=$(xcrun simctl list devices "${device_name}" available | grep "${device_name} (")
+          re='\(([-0-9A-Fa-f]+)\)'
+          [[ $device =~ $re ]] || exit 1
+          xcodebuild -workspace PagerViewExample.xcworkspace -UseNewBuildSystem=YES -scheme PagerViewExample -destination "platform=iOS Simulator,id=${BASH_REMATCH[1]}" CODE_SIGNING_ALLOWED=NO ONLY_ACTIVE_ARCH=YES COMPILER_INDEX_STORE_ENABLE=NO build
```
Comment (krozniata) [inline .github/workflows/ios.yml:62]:
```suggestion
        run: xcodebuild -scheme PagerViewExample -workspace PagerViewExample.xcworkspace -sdk iphonesimulator -configuration Debug
```

**Review by krozniata — APPROVED** [review]

**Review by troZee — APPROVED** [review]

**Pull request merged**